[DOCS] Add intro to traces (#1859)
grafana/tempo: docs/tempo/website/_index.md

<p align="center"><img src="logo_and_name.png" alt="Tempo Logo"></p>

> Grafana Tempo is an open source, easy-to-use and high-volume distributed tracing backend. Tempo is cost-efficient, requiring only object storage to operate, and is deeply integrated with Grafana, Mimir, Prometheus, and Loki. Tempo can be used with any of the open-source tracing protocols, including Jaeger, Zipkin, and OpenTelemetry.

Reviewer: ```suggestion
Grafana Tempo is an open source, easy-to-use, and high-volume distributed tracing backend. Tempo is cost-efficient, requiring only object storage to operate, and is deeply integrated with Grafana, Mimir, Prometheus, and Loki. You can use Tempo with any of the open-source tracing protocols, including Jaeger, Zipkin, and OpenTelemetry.
```

Reviewer: > high-volume distributed tracing backend. Tempo is cost-efficient, requiring only object storage to operat

In this case, what's the distinction between calling it `backend` and just calling it a database? Database seems more precise (too precise?)

Author: I asked the Tempo team wither backend or database was the more appropriate term. @mdisibio noted a preference for backend because database implies SQL, tables, etc. (which Tempo does not have).

> Grafana Tempo is an open source, easy-to-use, and high-volume distributed tracing backend. Tempo is cost-efficient, requiring only object storage to operate, and is deeply integrated with Grafana, Mimir, Prometheus, and Loki. You can use Tempo with any of the open-source tracing protocols, including Jaeger, Zipkin, and OpenTelemetry.

Reviewer 2: ```suggestion
Grafana Tempo is an open source, easy-to-use, and high-volume distributed tracing backend. Tempo is cost-efficient, and only requires an object storage to operate. Tempo is deeply integrated with Grafana, Mimir, Prometheus, and Loki. You can use Tempo with any of the open-source tracing protocols, including Jaeger, Zipkin, or OpenTelemetry.
```

Reviewer 2: Does a user use more than one of these protocols at the same time?

Reviewer 2: Alternative to keep the grammar in tact: "You can use Tempo with open-source tracing protocols, including Jaeger, Zipkin, and OpenTelemetry."

Author: Thank you. I've updated the text per your suggestions.

> Tempo has strong integrations with a number of existing open source tools, including: 

Reviewer 2: ```suggestion
Tempo integrates well with a number of existing open source tools: 
```

> - **Grafana**. Grafana ships with native support for Tempo using the built-in [Tempo data source](https://grafana.com/docs/grafana/latest/datasources/tempo/).

Reviewer 2: ```suggestion
- **Grafana** ships with native support for Tempo using the built-in [Tempo data source](https://grafana.com/docs/grafana/latest/datasources/tempo/).
```

> - **Grafana Loki**. Loki, with its powerful query language [LogQL v2](https://grafana.com/blog/2020/10/28/loki-2.0-released-transform-logs-as-youre-querying-them-and-set-up-alerts-within-loki/) allows us to filter down on requests that we care about, and jump to traces using the [Derived fields support in Grafana](https://grafana.com/docs/grafana/latest/datasources/loki/#derived-fields).

Reviewer 2: ```suggestion
- **Grafana Loki**, with its powerful query language [LogQL v2](https://grafana.com/blog/2020/10/28/loki-2.0-released-transform-logs-as-youre-querying-them-and-set-up-alerts-within-loki/) allows you to filter requests that you care about, and jump to traces using the [Derived fields support in Grafana](https://grafana.com/docs/grafana/latest/datasources/loki/#derived-fields).
```
Trying to come up with a new phrase besides "filter down on"...

> Tempo has strong integrations with a number of existing open source tools including 
> 
> - Grafana! Grafana ships with native support for Tempo using the built-in [Tempo data source](https://grafana.com/docs/grafana/latest/datasources/tempo/).
> - Grafana Loki - Loki, with its powerful query language [LogQL v2](https://grafana.com/blog/2020/10/28/loki-2.0-released-transform-logs-as-youre-querying-them-and-set-up-alerts-within-loki/) allows us to filter down on requests that we care about, and jump to traces using the [Derived fields support in Grafana](https://grafana.com/docs/grafana/latest/datasources/loki/#derived-fields).
> - Prometheus exemplars - which allow to jump from Prometheus metrics to Tempo traces by clicking on recorded exemplars. Read more about this integration in this [blog post](https://grafana.com/blog/2021/03/31/intro-to-exemplars-which-enable-grafana-tempos-distributed-tracing-at-massive-scale/).

Reviewer: ```suggestion
Tempo has strong integrations with a number of existing open source tools, including: 

- **Grafana.** Grafana ships with native support for Tempo using the built-in [Tempo data source](https://grafana.com/docs/grafana/latest/datasources/tempo/).
- **Grafana Loki**. Loki, with its powerful query language [LogQL v2](https://grafana.com/blog/2020/10/28/loki-2.0-released-transform-logs-as-youre-querying-them-and-set-up-alerts-within-loki/) allows us to filter down on requests that we care about, and jump to traces using the [Derived fields support in Grafana](https://grafana.com/docs/grafana/latest/datasources/loki/#derived-fields).
- **Prometheus exemplars**. Exemplars let you jump from Prometheus metrics to Tempo traces by clicking on recorded exemplars. Read more about this integration in this [blog post](https://grafana.com/blog/2021/03/31/intro-to-exemplars-which-enable-grafana-tempos-distributed-tracing-at-massive-scale/).
```

Reviewer: This format is the Google-recommended alternative to description lists:
https://developers.google.com/style/lists

Author: Thank you

> - **Prometheus exemplars**. Exemplars let you jump from Prometheus metrics to Tempo traces by clicking on recorded exemplars. Read more about this integration in this [blog post](https://grafana.com/blog/2021/03/31/intro-to-exemplars-which-enable-grafana-tempos-distributed-tracing-at-massive-scale/).

Reviewer 2: ```suggestion
- **Prometheus exemplars** let you jump from Prometheus metrics to Tempo traces by clicking on recorded exemplars. Read more about this integration in the blog post [Intro to exemplars, which enable Grafana Tempo’s distributed tracing at massive scale](https://grafana.com/blog/2021/03/31/intro-to-exemplars-which-enable-grafana-tempos-distributed-tracing-at-massive-scale/).
```

<p align="center"><img src="getting-started/assets/trace_custom_metrics_dash.png" alt="Trace visualization in Grafana "></p>

> Grafana Tempo builds an index on the high cardinality trace-id field and uses an object store as backend which allows for high parallelization of queries.

Reviewer: This reads ambiguous to me. What design feature makes it possible to have highly parallel queries, the index, the object store, or both?

Style-wise, should we hyphenate "high-cardinality" when used as an adjective?

Author: Yes, to high-cardinality as an adjective. Good catch. 

I asked the Tempo team about what enabled the highly parallel queries. Tempo is designed around a log-structure merge tree. Having blocks in object storage means it can query many blocks at once (thanks @mdisibio !).

> Grafana Tempo builds an index on the high cardinality trace-id field and uses an object store as backend which allows for high parallelization of queries.

Author: ```suggestion
Grafana Tempo builds an index on the high-cardinality trace-id field. Using an object store as backend lets Tempo query many blocks at once, so queries are highly parallelized. 
```

> Grafana Tempo builds an index on the high-cardinality trace-id field. Using an object store as backend lets Tempo query many blocks at once, so queries are highly parallelized. 

Reviewer 2: ```suggestion
Grafana Tempo builds an index from the high-cardinality trace-id field. Because Tempo uses an object store as a backend, Tempo can query many blocks simultaneously, so queries are highly parallelized. 
```
Vet my change for technical accuracy.

> Read more about this [architecture](https://grafana.com/docs/tempo/latest/operations/architecture/) documentation.

Reviewer 2: ```suggestion
For more information, see [Architecture](https://grafana.com/docs/tempo/latest/operations/architecture/).
```
